plugins/genesys/fu-genesys-usbhub-firmware.c

```diff
@@ -144,27 +152,68 @@ fu_genesys_usbhub_firmware_verify(const guint8 *buf, gsize bufsz, guint16 code_s
 	return TRUE;
 }
 
+gboolean
+fu_genesys_usbhub_firmware_query_codesize(FuFirmware *firmware,
+					  const guint8 *buf,
+					  gsize bufsz,
+					  GError **error)
+{
+	guint8 kbs = 0;
+	if (!fu_memread_uint8_safe(buf, bufsz, GENESYS_USBHUB_CODE_SIZE_OFFSET, &kbs, error)) {
+		g_prefix_error(error, "failed to get codesize: ");
+		return FALSE;
+	}
+	fu_firmware_set_size(firmware, 1024 * kbs);
+	return TRUE;
+}
+
+gboolean
+fu_genesys_usbhub_firmware_query_version(FuFirmware *firmware,
+					 const guint8 *buf,
+					 gsize bufsz,
+					 GError **error)
+{
+	guint16 version_raw = 0;
+	g_autofree gchar *version = NULL;
+
+	if (!fu_memread_uint16_safe(buf,
+				    bufsz,
+				    GENESYS_USBHUB_VERSION_OFFSET,
+				    &version_raw,
+				    G_LITTLE_ENDIAN,
+				    error)) {
+		g_prefix_error(error, "failed to get version: ");
+		return FALSE;
+	}
+	fu_firmware_set_version_raw(firmware, version_raw);
+	version =
+	    g_strdup_printf("%02x.%02x", (version_raw & 0xFF00U) >> 8, (version_raw & 0x00FFU));
+	fu_firmware_set_version(firmware, version);
+
+	/* success */
+	return TRUE;
+}
+
 static gboolean
 fu_genesys_usbhub_firmware_check_magic(FuFirmware *firmware,
 				       GBytes *fw,
 				       gsize offset,
 				       GError **error)
 {
-	guint8 magic[4] = {0x0};
+	guint8 magic[GENESYS_USBHUB_FW_SIG_LEN] = {0x0};
 
 	if (!fu_memcpy_safe(magic,
 			    sizeof(magic),
 			    0, /* dst */
 			    g_bytes_get_data(fw, NULL),
 			    g_bytes_get_size(fw),
 			    offset + GENESYS_USBHUB_FW_SIG_OFFSET,
-			    sizeof(magic),
+			    GENESYS_USBHUB_FW_SIG_LEN,
```

Comment:
nah, keep this as the sizeof(magic) please -- it's safer in case the buffer size changes or we move to a pointer passed in.

Reply:
searched related using and done.